shader/coarse.wgsl

```diff
@@ -310,7 +306,20 @@ fn main(
                 let blend = scene[dd];
                 is_blend = blend != BLEND_CLIP;
             }
-            let include_tile = tile.segment_count_or_ix != 0u || (tile.backdrop == 0) == is_clip || is_blend;
+
+            let di = draw_monoids[drawobj_ix].info_offset;
+            let draw_flags = info_bin_data[di];
+            let even_odd = (draw_flags & DRAW_INFO_FLAGS_FILL_RULE_BIT) != 0u;
+            let n_segs = tile.segment_count_or_ix;
+
+            // If this draw object represents an even-odd fill and we know that no line segment
+            // crosses this tile and then this draw object should not contribute to the tile if its
+            // backdrop (i.e. the winding number of its top-left corner) is even.
+            //
+            // NOTE: A clip should never get encoded with an even-odd fill.
+            let even_odd_discard = n_segs == 0u && even_odd && (abs(tile.backdrop) & 1) == 0;
```

Comment:
In my first pass, I was worried that this would be incorrect for even-odd clips. But I now see this can't be encoded, so my concern is more one of style than correctness.

If we ever do support even-odd clips (which is `eoclip` in PostScript and Core Graphics, `W*` in PDF), then we'd want to make the change. I think a version that gets this correct is just as concise, and potentially clearer, so suggest something along these lines:

```rust
    let backdrop_clear = select(tile.backdrop, tile.backdrop & 1, even_odd) == 0;
    let include_tile = n_segs != 0u || (backdrop_clear == is_clip) || is_blend;
```

But I wouldn't block the PR on this, it is correct as written.

Reply:
I like your formulation better than what I have as it is concise and doesn't rule out even-odd clips (which I wasn't aware of until now).